libopenage/util/fixed_point.h

```diff
@@ -464,16 +464,16 @@ class FixedPoint {
 		}
 
 		// Check for negative values
-		ENSURE(std::is_unsigned<raw_type>() or std::is_signed<raw_type>() and this->raw_value > 0, "FixedPoint::sqrt() is undefined for negative values.");
+		ENSURE(std::is_unsigned<raw_type>() or (std::is_signed<raw_type>() and this->raw_value > 0), "FixedPoint::sqrt() is undefined for negative values.");
```

Comment:
I think we could also simplify this with a `constexpr` if-clause to check the type at compile time. Then the ENSURE macro is only checked if `raw_type` is signed. Saves us a few calls in debug mode too :)

```suggestion
		if constexpr (std::is_signed<raw_type>()) {
			ENSURE(this->raw_value > 0, "FixedPoint::sqrt() is undefined for negative values.");
		}
```

Reply:
Great idea, thanks!